gems/aws-sdk-core/lib/aws-sdk-core/plugins/client_metrics_send_plugin.rb

```diff
@@ -62,17 +64,27 @@ def call(context)
             call_attempt.x_amzn_request_id = headers["x-amzn-request-id"]
           end
           call_attempt.http_status_code = context.http_response.status_code
-          if e = resp.error
+          context.metadata[:current_call_attempt] = call_attempt
+          request_metrics.add_call_attempt(call_attempt)
+          resp
+        end
+      end
+
+      class ErrorHandler < Seahorse::Client::Handler
```

Comment:
Nice! This is a good way of fixing this.

Reply:
What was the context of the bug relating to the CSM? Was it not capturing errors?

Reply:
I think it wasn't capturing errors for 6 years and nobody complained (I guess since this feature is effectively not used)